http-clients/src/main/java/com/palantir/remoting/http/HeaderAccessUtils.java

```diff
@@ -22,27 +22,38 @@
 import java.util.Map;
 import java.util.TreeMap;
 
+/**
+ * Used to access headers in a case-insensitive manner. This is necessary for compatibility with OkHttp 3.3.0+ as
+ * it lower-cases header names whereas we use constants from {@link com.google.common.net.HttpHeaders} where header
+ * names are in Train-Case. This can be removed once {@link feign.Request} and {@link feign.Response} expose the
+ * headers as a map which is case-insensitive with respect to the key. com.netflix.feign:feign-core:8.18.0 will
+ * have it for the {@link feign.Response} headers due to https://github.com/Netflix/feign/pull/418.
+ */
 public final class HeaderAccessUtils {
     private HeaderAccessUtils() {}
 
     public static boolean caseInsensitiveContains(Map<String, Collection<String>> headers, String header) {
-        return caseInsensitiveGet(headers, header) != null;
+        for (String headerName : headers.keySet()) {
+            if (headerName.equalsIgnoreCase(header)) {
+                return true;
+            }
+        }
+        return false;
     }
 
     public static Collection<String> caseInsensitiveGet(Map<String, Collection<String>> headers, String header) {
         return caseInsensitiveMultimapOf(headers).get(header);
```

Comment:
sorry, didn't catch this earlier, but I think making this also short-circuit as soon as it finds the match is probably better given where these things get called.